p2p/swarm.go

```diff
@@ -627,25 +627,29 @@ func (s *swarm) startNeighborhood() error {
 }
 
 func (s *swarm) peersLoop() {
+	var moreLock uint32 = 0
 loop:
 	for {
 		select {
 		case <-s.morePeersReq:
 			s.lNode.Debug("loop: got morePeersReq")
-			go s.askForMorePeers()
+			if atomic.CompareAndSwapUint32(&moreLock, 0, 1) {
```

Comment:
if this is called only here a simple boolean is enough since the only wat to call it is by using the channel